File: AdventOfCodeJava/tests/com/adventOfCode/PassphraseValidatorTests.java
package com.adventOfCode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class PassphraseValidatorTests {
	@Test
	void PassphraseValidator_BasicValidate_SimpleValid() {
		// act
		boolean result = PassphraseValidator.basicValidate("aa bb cc dd ee");
		
		// assert
		assertTrue(result);
	}
	
	@Test
	void PassphraseValidator_BasicValidate_SimpleInvalid() {
		// act
		boolean result = PassphraseValidator.basicValidate("aa bb cc dd aa");
		
		// assert
		assertFalse(result);
	}
	
	@Test
	void PassphraseValidator_BasicValidate_AlmostInvalid() {
		// act
		boolean result = PassphraseValidator.basicValidate("aa bb cc dd aaa");
		
		// assert
		assertTrue(result);
	}
	
	@Test
	void PassphraseValidator_CompexValidate_Valid() {
		// act
		boolean result = PassphraseValidator.complexValidate("abcde fghij");
		
		// assert
		assertTrue(result);
	}
	
	@Test
	void PassphraseValidator_CompexValidate_SingleAnagramInvalid() {
		// act
		boolean result = PassphraseValidator.complexValidate("abcde xyz ecdab");
		
		// assert
		assertFalse(result);
	}
	
	@Test
	void PassphraseValidator_CompexValidate_ValidWhenLengthsDifferent() {
		// act
		boolean result = PassphraseValidator.complexValidate("a ab abc abd abf abj");
		
		// assert
		assertTrue(result);
	}
	
	@Test
	void PassphraseValidator_CompexValidate_ValidOnlyTwoCharacters() {
		// act
		boolean result = PassphraseValidator.complexValidate("iiii oiii ooii oooi oooo");
		
		// assert
		assertTrue(result);
	}
	
	@Test
	void PassphraseValidator_CompexValidate_AllAnagramsInvalid() {
		// act
		boolean result = PassphraseValidator.complexValidate("oiii ioii iioi iiio");
		
		// assert
		assertFalse(result);
	}
}

File: AdventOfCodeCSharp/AdventOfCodeCSharp.Tests/PassphraseValidatorTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AdventOfCodeCSharp.Tests
{
    [TestClass]
    public class PassphraseValidatorTests
    {
        [TestMethod]
        public void PassphraseValidator_SimpleValid()
        {
            // act
            var result = PassphraseValidator.BasicValidate("aa bb cc dd ee");

            // assert
            Assert.IsTrue(result);
        }

        [TestMethod]
        public void PassphraseValidator_SimpleInvalid()
        {
            // act
            var result = PassphraseValidator.BasicValidate("aa bb cc dd aa");

            // assert
            Assert.IsFalse(result);
        }

        [TestMethod]
        public void PassphraseValidator_AlmostInvalid()
        {
            // act
            var result = PassphraseValidator.BasicValidate("aa bb cc dd aaa");

            // assert
            Assert.IsTrue(result);
        }

        [TestMethod]
        public void PassphraseValidator_ComplexValidate_Valid()
        {
            // act
            var result = PassphraseValidator.ComplexValidate("abcde fghij");

            // assert
            Assert.IsTrue(result);
        }

        [TestMethod]
        public void PassphraseValidator_ComplexValidate_SingleAnagramInvalid()
        {
            // act
            var result = PassphraseValidator.ComplexValidate("abcde xyz ecdab");

            // assert
            Assert.IsFalse(result);
        }

        [TestMethod]
        public void PassphraseValidator_ComplexValidate_ValidWhenLengthsDifferent()
        {
            // act
            var result = PassphraseValidator.ComplexValidate("a ab abc abd abf abj");

            // assert
            Assert.IsTrue(result);
        }

        [TestMethod]
        public void PassphraseValidator_ComplexValidate_ValidOnlyTwoCharacters()
        {
            // act
            var result = PassphraseValidator.ComplexValidate("iiii oiii ooii oooi oooo");

            // assert
            Assert.IsTrue(result);
        }

        [TestMethod]
        public void PassphraseValidator_ComplexValidate_AllAnagramsInvalid()
        {
            // act
            var result = PassphraseValidator.ComplexValidate("oiii ioii iioi iiio");

            // assert
            Assert.IsFalse(result);
        }
    }
}
